src/map/mob.c

```diff
@@ -1900,15 +1900,52 @@ static int mob_ai_hard(int tid, int64 tick, int id, intptr_t data)
 	return 0;
 }
 
+/**
+ * Adds random options of a given options drop group into item.
+ * 
+ * @param item : item receiving random options
+ * @param options : Random Option Drop Group to be used
+ */
+static void mob_setdropitem_options(struct item *item, struct optdrop_group *options)
+{
+	nullpo_retv(item);
+	nullpo_retv(options);
+	
+	for (int i = 0; i < options->optslot_count; i++) {
+		if (rnd() % 10000 >= options->optslot_rate[i])
+			continue;
+
+		// count avoids a too long loop that would cause lag. if after 10 full iterations it still fails
+		// it'll stop at one random index in the next iteration.
+		int count = 10 * options->optslot[i].option_count + (rnd() % options->optslot[i].option_count);
+		int idx = 0;
+		while (count > 0 && rnd() % 10000 >= options->optslot[i].options[idx].rate) {
+			idx = (idx + 1) % options->optslot[i].option_count;
+			--count;
+		}
+
+		item->option[i].index = options->optslot[i].options[idx].id;
+
+		int min = options->optslot[i].options[idx].min;
+		int max = options->optslot[i].options[idx].max;
```

Comment:
if vars used only here or/and below, declaration must be here too

Reply:
@4144 that's not what I'm talking about, there's no reason to have those variables in first place.

Reply:
@hemagx doesn't it improve readability? I actually decided to make them because writting them 3 times would make that line below quite polluted and hard to read:

```
item->option[i].value = min + (rnd() % (max - min + 1));
```
vs:
```
item->option[i].value = options->optslot[i].options[idx].min + (rnd() % (options->optslot[i].options[idx].max - options->optslot[i].options[idx].min + 1));
```

I could use 2 lines too:
```
item->option[i].value = options->optslot[i].options[idx].min +
                        (rnd() % (options->optslot[i].options[idx].max - options->optslot[i].options[idx].min + 1));
````

Reply:
first way also should be faster (with min, max vars)

Reply:
https://github.com/HerculesWS/Hercules/pull/1012#discussion_r48851863
Haru did say its ok to do it for improve readability

I also after improve readability rather than performance sometimes,
because having a high readability scripts can save human's time